### PR TITLE
[Phase 1] LPS-173369 Remove Stream and Optional for module: search-web-api part 1

### PR DESCRIPTION
**Pull request by julschong**
resubmitted from #3438

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:search

**Comment by Kvale1733** [comment]
@julschong No regressions found.

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/search/bar/portlet/display/context/factory/SearchBarPortletDisplayContextFactory.java [hunk 1]
```diff
@@ -303,25 +303,28 @@ protected SearchScopePreference getSearchScopePreference(
 		Portlet headerSearchBarPortlet =
 			searchBarPrecedenceHelper.findHeaderSearchBarPortlet(themeDisplay);
 
-		if (headerSearchBarPortlet != null) {
-			Optional<PortletPreferences> headerPortletPreferencesOptional =
-				portletPreferencesLookup.fetchPreferences(
-					headerSearchBarPortlet, themeDisplay);
+		if (headerSearchBarPortlet == null) {
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/search/bar/portlet/display/context/factory/SearchBarPortletDisplayContextFactory.java:306]: @julschong Please also apply this pattern to `_getScopeParameterName` and `_getKeywordsParameterName`.

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/search/request/SearchResponseImpl.java [hunk 2]
```diff
@@ -75,8 +72,14 @@ public SearchSettings getSearchSettings() {
 		return _searchSettings;
 	}
 
-	public Optional<String> getSpellCheckSuggestionOptional() {
-		return SearchStringUtil.maybe(_hits.getCollatedSpellCheckResult());
+	public String getSpellCheckSuggestion() {
+		String collatedSpellCheckResult = _hits.getCollatedSpellCheckResult();
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/search/request/SearchResponseImpl.java:76]: @julschong You missed this comment https://github.com/liferay-core-infra/liferay-portal/pull/3438#discussion_r1113885664

**Review by dantewang — CHANGES_REQUESTED** [review]
@julschong Please see comment and resend, thanks!

**Pull request closed without merging**